repair.go

```diff
@@ -70,21 +72,30 @@ func repairBadIndexVersion(logger log.Logger, dir string) error {
 		if _, err := io.Copy(repl, broken); err != nil {
 			return wrapErr(err, d)
 		}
+
+		var merr tsdb_errors.MultiError
+
 		// Set the 5th byte to 2 to indicate the correct file format version.
 		if _, err := repl.WriteAt([]byte{2}, 4); err != nil {
-			return wrapErr(err, d)
+			merr.Add(wrapErr(err, d))
+			merr.Add(wrapErr(repl.Close(), d))
+			return merr.Err()
 		}
 		if err := repl.Sync(); err != nil {
-			return wrapErr(err, d)
+			merr.Add(wrapErr(err, d))
+			merr.Add(wrapErr(repl.Close(), d))
+			return merr.Err()
 		}
 		if err := repl.Close(); err != nil {
 			return wrapErr(err, d)
 		}
 		if err := broken.Close(); err != nil {
 			return wrapErr(err, d)
 		}
-		if err := renameFile(repl.Name(), broken.Name()); err != nil {
-			return wrapErr(err, d)
+		if err := fileutil.Rename(repl.Name(), broken.Name()); err != nil {
+			merr.Add(wrapErr(err, d))
+			merr.Add(os.RemoveAll(repl.Name()))
```

Comment:
shouldn't repl be removed after any error 
```
repl, err := os.Create(filepath.Join(d, "index.repaired"))
if err != nil {
	return wrapErr(err, d)
}

defer func() {
	os.RemoveAll(repl.Name())
}()
```

Reply:
fixed